TimberAPIExample/Examples/EntityLinkerExample/UI/EntityLinkViewFactory.cs

```diff
@@ -0,0 +1,97 @@
+﻿using TimberbornAPI.Common;
+using TimberbornAPI.UIBuilderSystem;
+using UnityEngine;
+using UnityEngine.UIElements;
+using static UnityEngine.UIElements.Length.Unit;
+
+namespace TimberAPIExample.Examples.EntityLinkerExample.UI
+{
+    /// <summary>
+    /// Factory for creating small UI views for existing Links
+    /// </summary>
+    public class EntityLinkViewFactory
+    {
+        protected readonly UIBuilder _builder;
+
+        public EntityLinkViewFactory(
+            UIBuilder builder)
+        {
+            _builder = builder;
+        }
+
+        /// <summary>
+        /// Creates a basic link view with a image, label and delete button
+        /// </summary>
+        /// <param name="buttonLabelText"></param>
+        /// <returns></returns>
+        public virtual VisualElement Create(string buttonLabelText)
+        {
+            var rootBuilder =
```

Comment:
Is it possible to build each subcomponent separately, and then put them together at the end? Instead of all nested like this

Reply:
Yes for now it's smart to only use the builder once and with the addcomponent use the inbuilt builder. Or seperate them within methodes to make cleaner. This can be futher improved maybe when Builder is updated with other types of presets?

Reply:
Separated component creation a little to hopefully make it clearer. The UI stuff will be made better when there's time to first create the presets